partiql-planner/src/test/kotlin/org/partiql/planner/TyperTest.kt

```diff
@@ -0,0 +1,236 @@
+package org.partiql.planner
```

Comment:
Had some broader questions for porting the typing tests since I may be missing some context.
1. Is the goal here to eventually delete the `PlanTyperTestsPorted` (and related files)?
2. Where will the ported typing tests eventually live? In the conformance test suite?
3. At what point will we delete the existing `PlanTyperTestsPorted` (and related files)?
4. For future plan typing tests we want to add, what's the process for adding a test? Just add the test to the `plan-typer-tests-ported.ion` file?

Reply:
1. Yes.
2. I'd personally like to extract these interfaces into another package: `test/partiql-test-interfaces` that can be imported as a test dependency across the project. Any particular subproject may have local serialized tests -- and once the team agrees upon the tests' correctness, they can be moved into the conformance test suite. If you'd like, I can move these interfaces immediately in this PR.
3. The only open action item for `PlanTyperTestsPorted` has to do with the serialization of errors, which is being worked on. I actually don't **need** to keep it right now. I can always resurface the `PrintPlanTyperTestsPorted` from an older commit to re-serialize when the error serialization is addressed.
4. You can add a test to the `tests` directory in any file you'd like.